src/loaders/OmeZarrLoader.ts

```diff
@@ -285,10 +287,14 @@ class OMEZarrLoader implements IVolumeLoader {
     for (let i = 0; i < displayMetadata.channels.length; ++i) {
       chnames.push(displayMetadata.channels[i].label);
     }
+
+    // get shape of level 0
+    const shape0 = await fetchShapeOfLevel(store, loadSpec.subpath, datasets[0]);
```

Comment:
Curious what the overhead of this function is (`async` because it calls zarr `openArray`). How much work does `openArray` do? Does it have a cache to work with?

Reply:
Yep, this is another small network round trip to get the shape info.  With ome-zarr you basically have to do this once per level if you want to get all the array dims for everything.  Pretty minor though, and negligible compared to the pixel data requests that come after.

Reply:
I doubt that the zarr.js lib does any internal caching of its own.  Would be amazing if it did.